src/Connection.php

```diff
@@ -115,6 +122,8 @@ public function __construct(
         protected Driver $driver,
         ?Configuration $config = null,
     ) {
+        $this->heartbeat = $params['check_connection_frequency'] ?? null;
```

Comment:
I don't think `params` is the right place for this: https://github.com/doctrine/dbal/blob/c70bae9e69e0e543044d5e282a624d9b3cbea6fb/src/Connection.php#L80

Maybe the `Configuration` object is better for this?

Reply:
updated